org.freecad.FreeCAD.yaml

```diff
@@ -0,0 +1,802 @@
+      - -DHDF5_BUILD_CPP_LIB=ON
+      - -DHDF5_ENABLE_PARALLEL=ON
+      - -DHDF5_ENABLE_Z_LIB_SUPPORT=ON
+    cleanup:
+      - /bin
+      - /cmake
+    sources:
+      - type: archive
+        url: https://github.com/HDFGroup/hdf5/archive/hdf5-1_12_3.tar.gz
+        sha256: 39e2e3e25f2263f2dc3370556c174952675d7ac51c4d6f7d4f5e8b3a8bd10ac6
+
+  - name: med
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+    cleanup:
+      - /bin
+      - /share/doc
+    sources:
+      - type: archive
+        url: http://files.salome-platform.org/Salome/medfile/med-4.1.1.tar.gz
+        sha256: a082b705d1aafe95d3a231d12c57f0b71df554c253e190acca8d26fc775fb1e6
+      - type: patch
+        path: patches/med-hdf5-1.12.patch
+
+  - name: libspnav
+    sources:
+      - type: archive
+        url: https://github.com/FreeSpacenav/libspnav/releases/download/v1.1/libspnav-1.1.tar.gz
+        sha256: 04b297f68a10db4fa40edf68d7f823ba7b9d0442f2b665181889abe2cea42759
+
+  - name: boost
+    buildsystem: simple
+    build-commands:
+      - ./bootstrap.sh --prefix=/app --with-libraries=filesystem,program_options,regex,system,thread,date_time,chrono,atomic
+      - ./b2 install -j $FLATPAK_BUILDER_N_JOBS
+    sources:
+      - type: archive
+        url: https://boostorg.jfrog.io/artifactory/main/release/1.84.0/source/boost_1_84_0.tar.bz2
+        sha256: cc4b893acf645c9d4b698e9a0f08ca8846aa5d6c68275c14c3e7949c24109454
+
+  - name: coin
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+    cleanup:
+      - /bin
+    sources:
+      - type: archive
+        url: https://github.com/coin3d/coin/releases/download/v4.0.3/coin-4.0.3-src.tar.gz
+        sha256: 66e3f381401f98d789154eb00b2996984da95bc401ee69cc77d2a72ed86dfda8
+
+  - name: soqt
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DSOQT_BUILD_DOCUMENTATION=OFF
+    sources:
+      - type: git
+        url: https://github.com/coin3d/soqt.git
+        commit: 17364833049d3855f5e9e0f7908eb5ba08209318 # tag v1.6.2
+
+  - name: pivy
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DPIVY_USE_QT6=ON
+    sources:
+      - type: archive
+        url: https://github.com/coin3d/pivy/archive/refs/tags/0.6.9.a0.tar.gz
+        sha256: 2c2da80ae216fe06394562f4a8fc081179d678f20bf6f8ec412cda470d7eeb91
+      - type: shell
+        commands:
+          - sed -i 's|${Python_SITEARCH}|/app/lib/python3.11/site-packages/|g' {*/,./}CMakeLists.txt
+
+  - name: Cython
+    buildsystem: simple
+    build-commands:
+      - python3 setup.py install --prefix=/app --root=/
+    sources:
+      - type: archive
+        url: https://files.pythonhosted.org/packages/2a/97/8cc3fe7c6de4796921236a64d00ca8a95565772e57f0d3caae68d880b592/Cython-0.29.37.tar.gz
+        sha256: f813d4a6dd94adee5d4ff266191d1d95bf6d4164a4facc535422c021b2504cfb
+
+  - name: numpy
```

Comment:
why are these being manually generated ? calling setup.py is deprecated. why not use pip-generator ?

Reply:
ignornace I guess, is there any docs about the proper way of doing this? and again this is coming straight from the current flatpak. https://github.com/flathub/org.freecadweb.FreeCAD/blame/105074876d9b26ab0469d6b68589faeeff2fd59a/org.freecadweb.FreeCAD.yaml#L235C11-L246

Reply:
that's what the original package does. It probably predate all that tooling, the original maintainer went MIA a while back so I just maintained it.

Reply:
https://github.com/flatpak/flatpak-builder-tools/tree/master/pip#flatpak-pip-generator point it to your requirements.txt or individual modules.

The ones below are generated with pip-generator.

Reply:
I use the tool for recent dependencies, but the whole thing is fragile. I'd leave it like that until necessary.

Numpy is a pain to build.